client/pages/blogview.jsx

```diff
@@ -4,18 +4,66 @@ import { format } from 'date-fns';
 class BlogView extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { post: null };
+    this.state = {
+      post: null
+    };
+    this.toggleLike = this.toggleLike.bind(this);
   }
 
   componentDidMount() {
+    this.getLikes();
+  }
+
+  getLikes() {
     fetch(`/api/posts/${this.props.postId}`)
       .then(res => res.json())
-      .then(post => this.setState({ post }));
+      .then(postInfo => {
+        fetch(`api/likes/${this.props.postId}`)
+          .then(res => res.json())
+          .then(likes => {
+            const { totalLikes, userLiked } = likes;
+            const post = { ...postInfo, totalLikes, userLiked };
+            this.setState({ post });
+          });
+
+      })
+      .catch(err => console.error(err));
+  }
+
+  toggleLike() {
+    const { postId, userId, userLiked } = this.state.post;
+    const newLike = {
+      postId: postId,
+      userId: userId
+    };
+
+    if (userLiked) {
+      const req = {
+        method: 'DELETE'
+      };
+      fetch(`/api/likes/${postId}`, req)
+        .catch(err => console.error(err));
+      this.getLikes();
```

Comment:
Instead of re-fetching the likes, you can update the likes you have already loaded. When possible, it's better to avoid re-fetching data so that:
- the user doesn't have to wait as long if the server takes time to respond
- the server doesn't have to handle as many requests